src/components/atoms/ProjectSearchHeadContent/ProjectSearchHeadContent.tsx

```diff
@@ -45,34 +50,55 @@ export const ProjectSearchHeadContent = (props: ProjectSearchHeadContentProps) =
+      <RemoveCompareIcon onClick={() => handleProjectRemovedFromCompare(project.warehouseProjectId)} />
+    ) : (
+      <AddCompareIcon onClick={() => handleProjectAddedToCompare(project)} />
+    )
+  }
 
   const generateTableRow = (projectList: ProjectSearchResult[]) => {
-    return projectList.map((projectResults, idx) => (
-      <Tr
-        onClick={(event) => handleClick(projectResults.warehouseProjectId, event)}
-        onMouseEnter={() => changeHoverColor(`project-row-${idx}`, `hoverGreen`)}
-        data-testid="project-search-head-row"
-        className={`project-row-${idx}`}
-        key={`project-row-${projectResults.warehouseProjectId}`}
-        height="92px"
-      >
-        <Td data-testid="project-search-head-row-td">
-          <Flex alignItems="center">
-            <Box title={projectResults.name} overflow="hidden" flex={1}>
-              <Text fontWeight={500}>{projectResults.name}</Text>
-              <Text fontSize="sm">{projectResults.id}</Text>
-              <Text textOverflow="ellipsis" color="lightGray.700" fontSize="sm">
-                {projectResults.developer}
-              </Text>
-            </Box>
-            <Menu variant="menuWhite">
-              <MenuButton as={Button} textAlign="center" iconSpacing={0} rightIcon={<KebabMenuIcon />} variant="lightGrayRound32"></MenuButton>
-              <MenuContent menuItems={generateMenuList(projectResults.warehouseProjectId)} />
-            </Menu>
-          </Flex>
-        </Td>
-      </Tr>
-    ))
+    return projectList.map((projectResults, idx) => {
+      return (
+        <Tr
+          onClick={(event) => handleClick(projectResults.warehouseProjectId, event)}
+          onMouseEnter={() => changeHoverColor(`project-row-${idx}`, `hoverGreen`)}
+          data-testid="project-search-head-row"
+          className={`project-row-${idx}`}
+          key={`project-row-${projectResults.warehouseProjectId}`}
+          height="92px"
+        >
+          <Td data-testid="project-search-head-row-td">
+            <Flex alignItems="center">
+              <Box title={projectResults.name} overflow="hidden" flex={1}>
+                <Text fontWeight={500}>{projectResults.name}</Text>
+                <Text fontSize="sm">{projectResults.id}</Text>
+                <Text textOverflow="ellipsis" color="lightGray.700" fontSize="sm">
+                  {projectResults.developer}
+                </Text>
+              </Box>
+              {isCompare ? (
+                renderAddRemoveCompareButton(projectResults)
+              ) : (
+                <Menu variant="menuWhite">
+                  <MenuButton as={Button} textAlign="center" iconSpacing={0} rightIcon={<KebabMenuIcon />} variant="lightGrayRound32"></MenuButton>
+                  <MenuContent menuItems={generateMenuList(projectResults.warehouseProjectId)} />
+                </Menu>
+              )}
+            </Flex>
+          </Td>
+        </Tr>
+      )
+    })
```

Comment:
Shouldn't it be moved to its own component?

Reply:
I feel may be because component is rendering a table, all render row logic was also kept in the same component as its not reused anywhere else.

Reply:
I think its good to be like this but this has to be under organisms not atoms